lua/tree.lua

```diff
@@ -112,14 +116,15 @@ local function is_file_readable(fname)
 end
 
 function M.find_file(with_open)
-  local bufname = api.nvim_buf_get_name(api.nvim_get_current_buf())
-  if not is_file_readable(bufname) then return end
+  local bufname = vim.fn.bufname()
+  local filepath = vim.fn.fnamemodify(bufname, ':p')
```

Comment:
I wonder if there are two files with the same name in different folders, what would happen ?
why is `nvim_buf_get_name` not working ?

Reply:
Running `bufname` with no arguments defaults to the current buffer, if you run `fnamemodify` on two files with the same name the full path for each will be resolved, because each time you are calling `bufname('%')` essentially. Basically vim won't get mixed up it will only be the *current* buffer at the time that function is called, not sure if that answers your question.

Re. why `nvim_buf_get_name` isn't working it returns a different filename than expected. When testing locally. I logged the result of `vim.fn.bufname` and `api.nvim_buf_get_name`, say I am in `tree.lua` and I hit toggle and that function runs, `bufname` returns `tree.lua` and `api.nvim_buf_get_name` returns `my/full/path/LuaTree`, I'm not sure why the api behaves differently here but my understanding is that we need the *current* file name here i.e. the one we called this function from so it should be `tree.lua` not `LuaTree` since that's not the buffer where the function was called

Reply:
I should note that *most* of the time `api.nvim_buf_get_name` behaves as expected but not in this case. I think there's some extra logic that makes it different in some cases somehow but I don't know why.

Reply:
well i guess its some kind of priority stuff between lua and vim but not sure. We'll see that another time :)